ci/component.yml

```diff
@@ -5,11 +5,17 @@ platform: linux
 image_resource:
   type: docker-image
   source:
-    repository: golang
-    tag: 1.17.6
+    repository: taylorsilva/dcind
```

Comment:
do we have a list of approved containers to run in our CI?

Reply:
I don't think there's an official list anywhere, might be an idea to raise it with someone

Reply:
same comment as below

Reply:
the purpose of this image is to be able to run docker in docker in the CI.

Reply:
weve always used the golang repo up until this point but as this is a new use case it's worth bringing to the attention of the relevant guild or wider dev team and sharing that w're using this for X use case. Dont think there's any need to block this PR over it, but good to share